examples/evm-price-feed/src/feeds/binance.rs

```diff
@@ -0,0 +1,26 @@
+use anyhow::Result;
+
+const API_URL: &str = "https://api.binance.com/api/v3/ticker/price?symbol=";
+
+/// Fetches the current price for a trading pair from Binance API
+///
+/// # Arguments
+/// * `symbol_a` - The base symbol (e.g., "BTC")
+/// * `symbol_b` - The quote symbol (e.g., "USD")
+/// * `decimals` - The number of decimal places for the returned price
+///
+/// # Returns
+/// The price as u128 with the specified decimal precision
+pub fn fetch_token_price(symbol_a: &str, symbol_b: &str, decimals: u32) -> Result<u128> {
+    // Validate input symbols
+    if symbol_a.is_empty() || symbol_b.is_empty() {
+        return Err(anyhow::anyhow!(
+            "Invalid symbols: '{}' and '{}'",
```

Comment:
Prefer inlining the variable names in the template: ` "Invalid symbols: '{symbol_a}' and '{symbol_b}'",` unless you have to do modifications/accessors on the variables.

Rust clippy/fmt has started complaining about that on later versions for the `format` macro, I suppose it doesn't for `anyhow`, but for consistency it would be nice.